app/controllers/admin/merchants_controller.rb

```diff
@@ -13,15 +13,24 @@ def edit
 
   def update
     merchant = Merchant.find(params[:id])
-    merchant.update!(merchant_params)
-    flash[:notification] = 'Merchant successfully updated!'
 
-    redirect_to admin_merchant_path(merchant.id)
+    if params[:status]
+      merchant.update!(status_param)
+      redirect_to admin_merchants_path
+    else
+      merchant.update!(merchant_params)
+      flash[:notification] = 'Merchant successfully updated!'
+      redirect_to admin_merchant_path(merchant.id)
+    end
   end
 
   private
 
   def merchant_params
     params.require(:merchant).permit(:name)
   end
+
+  def status_param
+    params.permit(:status)
+  end
```

Comment:
Is the status_param method for security purposes? Similar to the merchant_params? This way one someone updates a merchant status they cannot inspect and pass any unwanted data?

Reply:
TBH, I don't know. I just did `params.permit` because that's what we did for other params.